bindings/java/src/main/com/apple/foundationdb/FDBTransaction.java

```diff
@@ -756,12 +757,17 @@ protected long getPtr() {
 
 	@Override
 	protected void finalize() throws Throwable {
+		long start = System.nanoTime();
 		try {
 			checkUnclosed("Transaction");
 			close();
 		}
 		finally {
 			super.finalize();
+			long end = System.nanoTime();
+			if(eventKeeper!=null){
+				eventKeeper.time(EventKeeper.Events.TRANSACTION_FINALIZER_TIME_NANOS, end-start, TimeUnit.NANOSECONDS);
```

Comment:
Is there a reason to focus on transaction finalizers in particular, and not all finalizer calls? Is it because a `Transaction` has a ready event keeper?

I also think that it would probably be good to differentiate between finalizer calls where the transaction was already closed versus ones where the transaction needed to be closed in the finalizer (i.e., was leaked)